xlators/storage/posix/src/posix-common.c

```diff
@@ -1231,6 +1281,106 @@ posix_init(xlator_t *this)
+                         "tt", rename_gfid_str);
+                snprintf(gfid_path, sizeof(gfid_path), "%s/%02x/%02x/%s",
+                         dir_handle, rename_gfid[0], rename_gfid[1],
+                         rename_gfid_str);
+                ret = sys_readlink(rename_gfid_path, link_path,
+                                   sizeof(link_path));
+                if (ret != -1) {
+                    /* link_path will be like this
+                       ../../<fhash>/tt/gfid_str so to populate
+                       absolute path from link path move the base_ptr to 6
+                       position and form an abosolute path and then read
+                      gfid from a absolute path and compare with
+                      rename_gfid
+                    */
+                    snprintf(abs_path, sizeof(abs_path), "%s/%s", dir_handle,
+                             (link_path + 6));
+                    size = sys_lgetxattr(abs_path, "trusted.gfid", abs_gfid,
+                                         16);
+                    if (size == 16) {
+                        if (!gf_uuid_compare(abs_gfid, rename_gfid)) {
+                            posix_handle_unset(this, rename_gfid, NULL);
+                            unlink_flag = 0;
+                            ret = sys_rename(rename_gfid_path, gfid_path);
+                            if (ret) {
+                                gf_log(this->name, GF_LOG_ERROR,
+                                       "rename gfid %s to expected gfid %s "
+                                       "is failed",
+                                       rename_gfid_path, gfid_path);
+                            } else {
+                                gf_log(this->name, GF_LOG_INFO,
+                                       "rename gfid %s to expected gfid %s "
+                                       "is passed",
+                                       rename_gfid_path, gfid_path);
+                            }
+                        }
+                    } else {
+                        gf_log(this->name, GF_LOG_INFO,
+                               "rename was not happened during crash so "
+                               "unlink %s path",
+                               rename_gfid_path);
+                        ret = 0;
+                    }
+                } else {
+                    ret = 0;
+                }
+            }
+
+            if (unlink_flag) {
+                snprintf(unlink_gfid, sizeof(unlink_gfid), "%s/%s", "tt",
+                         dentry->d_name);
+                ret = sys_unlinkat(_private->arrdfd[i], unlink_gfid);
+                if (ret) {
+                    gf_log(this->name, GF_LOG_INFO,
+                           "Unlink entry %s is failing in directory %s",
+                           unlink_gfid, dir_name);
+                }
+            }
+        }
+        sys_closedir(drptr);
```

Comment:
This loooong piece of code makes it hard to understand. Could you split long pieces of code like this into smaller functions ? It will reduce the indentation significantly, making it easier to read. It will also improve understandability by having meaningful function names that will help explain what the code is doing.

Reply:
done